server/users/controller.js

```diff
@@ -18,17 +18,19 @@ export const getUsers = async (userQuery) => {
     });
   });
 
-  console.log(userList);
   return userList;
 };
 
+export const getUserById = async (userID) => {
+  const doc = await db.collection(USERS).doc(userID).get();
+  return doc.data();
+};
+
 // add users to firestore
 export const addUser = async (user) => {
   const newUserID = user.uid; // from firebase. Will replace autogenerated id created by firestore for new documents.
-  const newUser = { username: user.username, email: user.email };
+  const newUser = { username: user.username, email: user.email, id: newUserID, image: user.picture };
 
   // https://firebase.google.com/docs/firestore/manage-data/add-data
-  await db.collection(USERS).doc(newUserID).set(newUser);
-
-  console.log(newUser);
+  return db.collection(USERS).doc(newUserID).set(newUser);
```

Comment:
will this add an additional Id field to the firestore document? `.doc(newUserID)` changes the default id to `newUserID` right? But then your also passing newUser which has `id` as a field. Maybe it won't add it since it is the same field name. So then do we need to have it as a field in newUser?

Reply:
I want to add `id` on the user object as well - this way when we load a list of users, or attach an individual user object, the `id` field is there for us to use

Reply:
oh I see what you are saying. So we can't access the internal id field used by firestore? 
![image](https://user-images.githubusercontent.com/35232697/107888736-cbb93700-6ed3-11eb-8c0c-2df92b0a53a4.png)

It seems to have it as a header but not a field.

Reply:
Depends on how - there are some ways, but it's simpler just to include it inside 

I'll update the quiz creation to use [this method to get the ID](https://firebase.google.com/docs/firestore/manage-data/add-data#add_a_document) too